### ISSUE TITLE
[WC 5.0 beta1 with WC Admin 1.8.3] Connection to woo.com doesn't work when trying to complete `Add paid extensions to my store` Home Screen task

### ISSUE THREAD

**Issue by juliaamosova** [woocommerce/woocommerce-admin#6103]
**Describe the bug**

Connection to woo.com doesn't work when trying to complete the `Add paid extensions to my store` Home Screen task. 

As long as the connection to woo.com is made before starting that `Add paid extensions to my store` task, it works. But if the connection to woo.com has to be done within that task for the first time, the task can't be completed.

Also, this seems to happen on Atomic only. I was able to reproduce it several times so as @claudiosanches when we were testing WC 5.0 beta1. 

**To Reproduce**

1. Create a brand new Atomic site.
2. During onboarding, make sure to select all additional paid products:

![Screen Shot 2021-01-19 at 5 36 32 PM](https://user-images.githubusercontent.com/19143190/105110133-f0b0ba80-5a8b-11eb-9bfd-d8f8d2a6ed6d.png)

3. Finish onboarding and navigate to Home Screen. Click on `Add paid extensions to my store` task:

![purchase_flow](https://user-images.githubusercontent.com/19143190/105110171-0920d500-5a8c-11eb-8772-c79c5a75fe9b.jpg)

4. Proceed with the purchase of all the extensions. It should be completed without issues.
5. Next, you will be offered to install extensions on your site - click on `Add to site` for any of the extensions.
6. See that the redirect back to the site won't work:

![](http://g.recordit.co/n9vRV4QcOG.gif)

**Expected behavior**

I expected to be redirected back to the site in step 6 with all the extensions installed on the site. 

**Screenshots**

Couple more screenshots:

![Screen Shot 2021-01-19 at 5 26 52 PM](https://user-images.githubusercontent.com/19143190/105110406-81879600-5a8c-11eb-8d0b-91988c32f51e.png)

![Screen Shot 2021-01-19 at 5 28 07 PM](https://user-images.githubusercontent.com/19143190/105110413-851b1d00-5a8c-11eb-97e8-f9326fde54ce.png)

**Desktop (please complete the following information):**

* OS: macOS Catalina 10.15.6
* Browser: Chrome
* Version: 87.0.4280.141 (Official Build) (x86_64)

**Additional context**

Try going to `/wp-admin/admin.php?page=wc-addons&section=helper` and connect to your woo.com account, then go back to the `Add paid extensions to my store` task and try going through it again - it will work this time.  

As long as the connection to woo.com is made before starting that `Add paid extensions to my store` task, it works. But if the connection to woo.com has to be done within that task for the first time, the task can't be completed.

**Comment by timmyc** [comment]
Based upon where this problem is happening, this issue might actually be in the woocommerce.com side of things. I will add it to the milestone to have someone investigate further and verify though.

**Comment by becdetat** [comment]
When I tested this with:

- Jurassic Ninja
- WooCommerce 5.0.0-beta.1
- with WooCommerce Admin 1.8.3, 1.9.0-beta-2, and mainline

Everything worked fine.

When I tested this with:

- Atomic ephemeral
- WooCommerce 5.0.0-beta.1
- with WooCommerce Admin 1.8.3 and mainline

I could reproduce this error.

I agree with @timmyc that this is an environmental issue probably on woocommerce.com's side.

**Comment by becdetat** [comment]
@juliaamosova Did you get prompted to authorise Jetpack? 
![image](https://user-images.githubusercontent.com/224531/105447181-6bade700-5cbf-11eb-916f-8550a88b70de.png)
I didn't see it on your recording and it didn't appear when I was able to reproduce it.

This would indicate that something on woocommerce.com's side is preventing the connection, which would be the cause of the "We couldn't connect to your site" error.

**Comment by becdetat** [comment]
The error is also happening on an ephemeral site with WooCommerce 4.9.1

**Comment by psealock** [comment]
Thanks for the confirmation @becdetat. With the problem being isolated to something happening with Atomic sites, we can say the problem lies in Atomic infrastructure, woocommerce.com, or in wc-admin but is too involved to solve in a timely manner before the next 5.0.0 beta release. I will remove this from the milestone, but increase the priority to high so it can be addressed in the next cycle.

**Comment by ilyasfoo** [comment]
I am unable to reproduce this on both an ephemeral site and a JN site with WooCommerce 5.0. In both of my tests, I had the "Authorize connection" screen which may have something to do with the issue as observed by @becdetat.

**Comment by mattyza** [comment]
> I wish I could reference the p2 post, or conversation that I saw this in, but I believe the success rate of the remote extension installation flow is sub-optimal. Going to loop in @mattyza for thoughts and any background on the tracking in place on the woocommerce.com side of things in this equation.
> 
> Regardless though, I do think this issue might be out of scope for the wc-admin project, but until we fully determine that, I'm going to remove from this dev cycle as there is no clear immediate action we can take.

Thanks @timmyc - one-click install success rates are [currently tracking at 60% on average](https://mc.a8c.com/tracks/trends/?eventname=woocommercecom_inapppurchase_install_succeeded%2Cwoocommercecom_inapppurchase_install_error%2Cwoocommercecom_inapppurchase_install_started&interval=day&startdate=20191030&stat=uniques).

Without looking into the code too deeply, an assumption here could be that the install is being attempted as a parallel operation to the connection to WCCOM, which would introduce a race condition.

@juliaamosova - if you feel this is potentially a WooCommerce.com issue, I'd be glad to discuss further, isolate, and resolve.

**Comment by juliaamosova** [comment]
Thanks everyone for looking into this issue here! 

>Thanks @timmyc - one-click install success rates are currently tracking at 60% on average.

>Without looking into the code too deeply, an assumption here could be that the install is being attempted as a parallel operation to the connection to WCCOM, which would introduce a race condition.

The above seems to provide some explanation of what could be happening. When I was testing WooCommerce 5.1 recently, the connection would work randomly as well. 

>@juliaamosova - if you feel this is potentially a WooCommerce.com issue, I'd be glad to discuss further, isolate, and resolve.

I don't know if this is a WooCommerce.com issue but it sounds like. I can move this issue to woocommerce.com repo to see if you could work on it at some point. Let me know.

**Comment by mattyza** [comment]
> I don't know if this is a WooCommerce.com issue but it sounds like. I can move this issue to woocommerce.com repo to see if you all could work on it at some point. Let me know.

Thank @juliaamosova. Yes, please do. It sounds like we'll want to inspect for a possible race condition here.

**Comment by juliaamosova** [comment]
>Thank @juliaamosova. Yes, please do. It sounds like we'll want to inspect for a possible race condition here.

Thanks @mattyza! I created the issue in `woocommerce.com` repository: https://github.com/Automattic/woocommerce.com/issues/9779. 

I believe this thread can now be closed here in WC Admin.
